lib/_http_client.js

```diff
@@ -169,6 +172,14 @@ function ClientRequest(input, options, cb) {
   if (options.timeout !== undefined)
     this.timeout = getTimerDuration(options.timeout, 'timeout');
 
+  if (options.signal) {
```

Comment:
Should we extract the value in a variable to be safe? Options could be changed later or the property could be a getter that throws afterwards